### DISCUSSION TITLE
Added support for PKL syntax highlighting in GitHub

### DISCUSSION THREAD

**Discussion by darvld** [apple/pkl#246] [General]
> [!NOTE]
> At time of writing, the PR is currently [pending review in the Linguist repo](https://github.com/github-linguist/linguist/pull/6718)

Hey everyone, @sgammon and I have been looking a lot into PKL in recent times so we decided to add support for syntax highlighting and language detection in GitHub's [Linguist](https://github.com/github-linguist/linguist/pull/6718) repo, which (if merged) will allow GitHub to correctly visualize `.pkl` files in the web viewer.

Some details like the color of the language can be adjusted before the feature goes live (let us know if it's necessary), so we thought it would be a good idea to open a related discussion. Happy to contribute more in the future, thank you for the great work!

**Comment by odenix** [comment]
Great to see! How will GitHub distinguish between Pkl and Python's Pickle, which already registered the `.pkl` extension?

**Reply by darvld** [reply]
Linguist appears to support heuristics for this, we're currently looking into that to avoid issues (it's actually a requirement for merging).

**Reply by darvld** [reply]
Following up on heuristics, i'll be adding the following regex: `new|class|function|import|module|(\w+\s*({|=))`, it should be enough to separate Pkl sources from Pickle data. False positives shouldn't be an issue either way, I don't believe inspecting the binary data from Pickle is a common use case, so accidental Pkl highlighting will likely go unnoticed.

**Comment by sgammon** [comment]
Hey all,

One qualification for language support to get merged in Linguist is enough demand. GitHub asks in their PR-based quiz if there are at least "hundreds of repos" with a new language present.

We may have a tough time proving this, though, due to the extension overlap with Python Pickle. It may help, then, if interested parties [upvote the PR](https://github.com/github-linguist/linguist/pull/6718) to show GitHub that Pkl has a strong community deserving of syntax highlighting.

**Comment by bioball** [comment]
Hey, thanks for the work here! We're stepping on each other's toes a bit, though. We're actually preparing our own PR for linguist that we plan on submitting soon. Also, we have a much more robust textmate grammar than the one that's in pkl-vscode: https://github.com/apple/pkl.tmbundle

The one in pkl-vscode is pretty bare bones, because pkl-vscode uses tree-sitter-pkl to do most of the heavy lifting.

**Reply by darvld** [reply]
That's good to hear! Support on Linguist from the Pkl team is certainly welcome, we could close the PR on Linguist so you guys can go ahead when you're ready.

**Comment by sgammon** [comment]
@bioball @stackoverflow mentioned this work on the PR; sorry about that. We are happy to defer to the Pkl team in all regards 😁

**Reply by bioball** [reply]
No worries! Love the enthusiasm
